### ISSUE TITLE
Add support for prediction columns in mosaic

### ISSUE THREAD

**Issue by seyuboglu** [robustness-gym/mosaic#40]
Include the ability to add prediction columns directly to testbenches for evaluating models.

Potential design:
prediction mixin --> ClassifierMixin [logits, probs, argmax -- moving between things]

Need to support more than classification (e.g. segmentation, text generation).

Linking the prediction to the task. 

One caveat for ClassifierMixin is supporting multi-label problems (i.e. imageA can be both class1 and class2)

**Comment by seyuboglu** [comment]
#46
